src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/VSSettings.cs

```diff
@@ -75,7 +75,7 @@ private bool ResetSolutionSettingsIfNeeded()
             // In the open PM UI scenario (no restore run), there is an asynchronous invocation of this code path. This changes ensures that
             // the synchronous calls that come after the asynchrnous calls don't do duplicate work.
             // That however is not the case for solution close and  same session close -> open events. Those will be on the UI thread.
-            if (_solutionSettings == null || !string.Equals(root, _solutionSettings.Item1))
+            if (_solutionSettings == null || !string.Equals(root, _solutionSettings.Item1, StringComparison.OrdinalIgnoreCase))
```

Comment:
Same as well, Path comparisons should be aware of the FS casing. 
We have a helper method, OsAware or something like that.